packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/mocks.tsx

```diff
@@ -15,16 +21,94 @@ export const MOCK_RESET_TOKEN = 'mockResetToken';
 export const MOCK_RECOVERY_KEY_ID = 'mockRecoveryKeyId';
 export const MOCK_KB = 'mockkB';
 
+// TODO: combine a lot of mocks with AccountRecoveryResetPassword
+const fxDesktopV3ContextParam = { context: 'fx_desktop_v3' };
+
+export const mockCompleteResetPasswordParams = {
+  email: MOCK_ACCOUNT.primaryEmail.email,
+  emailToHashWith: MOCK_ACCOUNT.primaryEmail.email,
+  token: '1111111111111111111111111111111111111111111111111111111111111111',
+  code: '11111111111111111111111111111111',
+  uid: MOCK_ACCOUNT.uid,
+};
+
+export const paramsWithSyncDesktop = {
+  ...mockCompleteResetPasswordParams,
+  ...fxDesktopV3ContextParam,
+};
+
+export const paramsWithMissingEmail = {
+  ...mockCompleteResetPasswordParams,
+  email: '',
+};
+
+export const paramsWithMissingCode = {
+  ...mockCompleteResetPasswordParams,
+  code: '',
+};
+
+export const paramsWithMissingEmailToHashWith = {
+  ...mockCompleteResetPasswordParams,
+  emailToHashWith: '',
+};
+
+export const paramsWithMissingToken = {
+  ...mockCompleteResetPasswordParams,
+  token: '',
+};
+
+export function mockUrlQueryData(
+  params: Record<string, string> = mockCompleteResetPasswordParams
+) {
+  const window = new ReachRouterWindow();
+  const data = new UrlQueryData(window);
+  for (const param of Object.keys(params)) {
+    data.set(param, params[param]);
+  }
+  return data;
+}
+
+class StorageDataMock extends StorageData {
+  public override persist(): void {
+    // no op
+  }
+  public override load(): void {
+    // no op
+  }
+}
+
 export const Subject = ({
   account,
-  history,
+  params,
 }: {
   account: Account;
-  history?: History;
-}) => (
-  <AppContext.Provider value={mockAppContext({ account })}>
-    <LocationProvider {...{ history }}>
-      <AccountRecoveryConfirmKey />
-    </LocationProvider>
-  </AppContext.Provider>
-);
+  params?: Record<string, string>;
+}) => {
+  const windowWrapper = new ReachRouterWindow();
+  const urlQueryData = mockUrlQueryData(params);
+
+  return (
+    <AppContext.Provider
+      value={mockAppContext({
+        account,
+        windowWrapper,
+        storageData: new StorageDataMock(windowWrapper),
+        urlQueryData,
+      })}
+    >
+      <LocationProvider>
+        <LinkValidator
+          linkType={LinkType['reset-password']}
+          viewName={'account-recovery-confirm-key'}
```

Comment:
Does this need the binding, or should it just be
```suggestion
          viewName='account-recovery-confirm-key'
```